domain/bakerystorage/service/config.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+
+	"github.com/go-macaroon-bakery/macaroon-bakery/v3/bakery"
+	"github.com/juju/errors"
+)
+
+type ConfigState interface {
```

Comment:
Can this just be called `State`? Or `BakeryState`? The the word `Config` is confusing to me here, it implies that these are part of some config passed into the state.

Also a comment on it would be good.

Reply:
Yes, as Simon says.

His recommendation is to rename to BakeryConfigState, which I like